cmds/dhclient/dhcp6client/packetSock.go

```diff
@@ -0,0 +1,155 @@
+package dhcp6client
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/mdlayher/dhcp6"
+	"golang.org/x/sys/unix"
+)
+
+type packetSock struct {
+	fd      int
+	ifindex int
+}
+
+var bcastMAC = []byte{255, 255, 255, 255, 255, 255}
+
+func NewPacketSock(ifindex int) (*packetSock, error) {
+	fd, err := unix.Socket(unix.AF_PACKET, unix.SOCK_DGRAM, int(swap16(unix.ETH_P_IPV6)))
```

Comment:
I assume you're using swap16 because you know you're on a little endian machine. This will immediately break on a big-endian machine. further, this may be the source of your protocol error :-) since you may not need to swap, I will test. This is not how you want to manipulate endian-ness anyway; see the binary package.